### PR TITLE
new interface for MountImage added

### PR DESCRIPTION
**Pull request by kunalkushwaha**
Interface added to mount image rootfs without any container.
image can be mounted in ReadOnly or ReadWrite mode.

Signed-off-by: Kunal Kushwaha <kunal.kushwaha@gmail.com>

### CODE REVIEW

**Comment by kunalkushwaha** [comment]
This PR is for https://github.com/containers/libpod/issues/1433.

@rhatdan Please see if this looks okay.
- Is options need to be checked here or functions that invoke this Interface should take care of that?

**Comment by rhatdan** [comment]
You need to update cmd/containers-storage to support this mount.  Then we  need tests to test this.

Not sure if we would want to support mounting the image read/write?  

@nalind WDYT?

File: store.go [hunk 1]
```diff
@@ -253,6 +253,10 @@ type Store interface {
 	// Wipe removes all known layers, images, and containers.
 	Wipe() error
 
+	// MountImage mounts an image to temp folder and returns the mount point.
+	// Options  supported are ro, rw.
```
Comment (vrothberg) [inline store.go]: Double whitespace between "Options" and "supported".

**Comment by rhatdan** [comment]
I can see mounting an image Read/Only mounting it read/write seems like a bad idea unless their is a upper layer. I don't think we would want the Image Storage modified at all.

@vrothberg @nalind WDYT?

**Comment by kunalkushwaha** [comment]
@rhatdan I got your point. 

Earlier I thought, storage lib should provide the simple API, so consumer of API can choose the mode. 

But with without additional layer, mounting image should not be allowed as it will change the image. For all other cases `Mount()` should be used. I will make changes for `ReadOnly`.

**Comment by kunalkushwaha** [comment]
I am little confused towards naming the API.
`MountImage()` by name do not suggest it will return `ReadOnly` mounted image.
- So currently it takes `mountOptions` but is ignored and set to `ReadOnly` in function.
- In future, if this function need to add support of ReadWrite Mode, no need to change the Interface Name.

- We can remote `mountOptions` by changing the interfaceName as `MountImageRO()` or `MountROImage()`

@rhatdan @vrothberg @nalind PTAL

File: store.go [hunk 2]
```diff
@@ -2473,6 +2478,32 @@ func (s *store) Version() ([][2]string, error) {
 	return [][2]string{}, nil
 }
 
+func (s *store) MountImage(id string, mountOptions []string) (string, error) {
+	// Currently only ReadOnly option is set
+	// and value passed to mountOptions is ignored.
+	mountOptions = []string{"ro"}
```
Comment (rhatdan) [inline store.go]: This should be appended on, we still want to pass in the mount options.  We should support NODEV, NOEXEC, NOSUID ...

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
We need tests on this.  I want to make sure that I can mount an image and then create a container based on that image, umount the container.  Mount the image multiple times, umount it multiple times and keep track of how many times it is mounted.

**Comment by kunalkushwaha** [comment]
Test cases failing.. mostly while creating new layer. This works fine if run by command.

`populate` function fails.

**Comment by rhatdan** [comment]
@kunalkushwaha Could you try to rebase and push.  TO see if this fixes the issue.

**Comment by kunalkushwaha** [comment]
@rhatdan This PR is stuck with this test failing issue. PTAL

**Comment by vrothberg** [comment]
Restarted the tests to check if the errors are flakes. Thanks for the pin!

**Comment by rhatdan** [comment]
@kunalkushwaha tests seem to be failing.

**Comment by vrothberg** [comment]
Friendly ping. Any updates?

**Comment by kunalkushwaha** [comment]
Sorry, was not able to work on this . I will start on this now and update.

**Comment by vrothberg** [comment]
Awesome, thanks a lot, @kunalkushwaha !

**Comment by rhatdan** [comment]
@kunalkushwaha needs a rebase.

**Comment by mrunalp** [comment]
@kunalkushwaha Could you rebase this?

**Comment by rhatdan** [comment]
@ashley-cui PTAL

**Comment by rhatdan** [comment]
@kunalkushwaha We need this PR to move forward, are you still working on it?  If not we will take it over.

**Comment by kunalkushwaha** [comment]
@rhatdan : @sujil02 asked me a few days back regarding this and had explained to him. 
I thought he is already working on this.

The major blocker on this for merging was test case failure. (https://github.com/containers/storage/pull/438#issuecomment-550970311)

I may not be able to pick it up very soon. So if anyone can pick it up, it will be nice.

**Pull request closed without merging**